opal/mca/common/ofi/common_ofi.c

```diff
@@ -87,19 +91,30 @@ OPAL_DECLSPEC int opal_common_ofi_init(void)
 {
     int ret;
 
+    /*
+     * Global refcnt and boolean are used to ensure we only initialize once
```

Comment:
This pattern is well established and doesn't need a comment about why its here.  However, you do need to explain how this fits in with the thread safety requirements of this function (and document those requirements in the function header).

Reply:
I think to make it easier for users to call this function, I'll add a new opal_mutex_t lock and lock around the whole function.

Reply:
why is that the right thing to do?  That has its own risks, especially around initialization of the mutex in the initialization function you're currently writing.  Just get the definition right.

Reply:
There's already a common mutex lock that is being used, I think it's fine to re-use that